### PR TITLE
add symmetric quant in softmax

### PR DESCRIPTION
**Pull request by chenfucn**
### Description

https://github.com/microsoft/onnxruntime/issues/14626

### Motivation and Context

https://github.com/microsoft/onnxruntime/issues/14626

### CODE REVIEW

File: onnxruntime/python/tools/quantization/operators/softmax.py [hunk 1]
```diff
@@ -80,6 +80,11 @@ def quantize(self):
         if self.quantizer.activation_qType == onnx.onnx_pb.TensorProto.UINT8:
             out_scale = 1 / 256.0
             out_zero_point = 0
+        elif self.quantizer.is_activation_symmetric:
+            # results are all greater or equal to 0, so we can only use
+            # half of the range
+            out_scale = 1 / 127.0
```
Comment (wejoncy) [inline onnxruntime/python/tools/quantization/operators/softmax.py:86]: Would it be 128.0? as we have closed interval [0,127]

File: onnxruntime/python/tools/quantization/operators/softmax.py [hunk 2]
```diff
@@ -80,6 +80,11 @@ def quantize(self):
         if self.quantizer.activation_qType == onnx.onnx_pb.TensorProto.UINT8:
             out_scale = 1 / 256.0
             out_zero_point = 0
+        elif self.quantizer.is_activation_symmetric:
```
Comment (wejoncy) [inline onnxruntime/python/tools/quantization/operators/softmax.py:83]: Do we need to support symmetric in QLinearSoftmax also?

Comment (chenfucn) [reply onnxruntime/python/tools/quantization/operators/softmax.py:83]: no, qlinearXXX is the result of Q node merging with others

**Comment by chenfucn** [comment]
AVX2 and AVX512 instructions have saturation problems:

https://www.intel.com/content/www/us/en/develop/documentation/onednn-developer-guide-and-reference/top/advanced-topics/nuances-of-int8-computations.html

You can turn on session option "session.x64quantprecision". With this, ORT will promote int8 to int16 in GEMM operations. This will preserve accuracy but has significant performance degradation. 

See
https://github.com/microsoft/onnxruntime/pull/12088

File: onnxruntime/python/tools/quantization/operators/softmax.py (hunk 1 above)
Comment (chenfucn) [reply onnxruntime/python/tools/quantization/operators/softmax.py:86]: the symmetric range is [-127,127], have to discard -128 to make is symmetric. This range has 254 unit intervals, half of it is 127 intervals. So the correct value is 1/127

**Review by yufenglee — APPROVED** [review]
:shipit:

**Review by wejoncy — APPROVED** [review]

**Pull request merged**